kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java

```diff
@@ -241,6 +246,37 @@ public static void assertPartitionOffsetsEqual(List<PartitionOffset> a, List<Par
         valueDeserializerClassName, new Properties(), validateContents);
   }
 
+  public static <T> void testWithRetry(Supplier<Boolean> testCondition, String errorMessage) {
+    testWithRetry(
+        testCondition,
+        errorMessage,
+        DEFAULT_EXP_BACKOFF_RETRIES,
+        DEFAULT_INITIAL_BACKOFF_MS,
+        TimeUnit.MILLISECONDS);
+  }
+
+  public static <T> void testWithRetry(
+      Supplier<Boolean> testCondition,
+      String errorMessage,
```

Comment:
Make errorMessage the last arg, and also accept error message args, so that the `String.format` is done inside this function: e.g.

```
testWithRetry(true, "my test is %s", "awesome");
```

Reply:
I'd prefer to avoid that for a couple of reasons.

First, varargs in multi-argument functions are something that almost never reads well - the only exceptions I can think of being SLF4J-style loggers and even they are practically always limited to less than 2 unrelated arguments.
Second, in order not to force users to pass `null` more often than they want, vararg methods require overloads, which I don't think we need here.
Third, if we want to get everything working with a single `String.format` call we will need to change the wording (leaving the vararg-ed template, which is the important part of the message, last), I think either escape the vararg template or concatenate it to the main template, and probably re-wrap the varargs anyway. Alternatively we can use two `String.format` calls, but I don't think that's any sort of win then.

So unless you're feeling strongly about this, can we avoid the varargs?

Reply:
I still think that the API would be cleaner as in:

```
  public static <T> void testWithRetry(
￼      Supplier<Boolean> testCondition,
      int numRetries,
      long initialBackoff,
      TimeUnit backoffTimeUnit,
      String errorMessage,
      Object... errorMessageArgs) {
    // ...
    Assert.fail(
        String.format(
            "%s. Failed after %d exponential backoff retries with %d ms initial backoff",
            String.format(errorMessage, errorMessageArgs),
            numRetries,
            backoffTimeUnit.toMillis(initialBackoff)));
```

(errorMessage, errorMessageArgs) is a common pattern, and it makes usage easier. But I'm not strong about it. Acknowledged.

Reply:
I'd still prefer to take advantage of your ack, because for the purpose of this method I can easily see how the error message might not have arguments (my initial error messages were simple strings), I hate either passing `null` in such cases or providing an overload that hides the `null`, and I'm afraid of call-sites being
```
testWithRetry(
    () -> {
        <epic lambda body here>
    },
    /* numRetries= */ 5,
    /* initialBackoff= */ Duration.ofMillis(500),
    "This %s was not that %s but a third %s",
    thisObject,
    thatObject,
    thirdObject);
```

Thank you for the enjoyable discussion :)

Reply:
> I hate either passing null in such cases or providing an overload that hides the null
Not important anymore, but be aware in the future that 0 arguments is a valid input a var-arg parameter.

`foo(String... bar)` can be called as just `foo()` (an empty array will be created and passed as argument.